src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -737,19 +740,34 @@ public function setAutowiringTypes(array $types)
      */
     public function isAutowired()
     {
+        return (bool) $this->autowired;
+    }
+
+    /**
+     * Gets the autowiring mode.
+     *
+     * @return int
+     */
+    public function getAutowired()
+    {
         return $this->autowired;
     }
 
     /**
      * Sets autowired.
      *
-     * @param bool $autowired
+     * @param int $autowired
```

Comment:
Imo we should either explicitly allow `false` or add a new constant, it feels weird to call `$definition->setAutowired(0);`.